src/codegen/array_boundary.rs

```diff
@@ -0,0 +1,159 @@
+    cfg: &mut ControlFlowGraph,
+    minus: bool,        // If the function is called from pushMemory or popMemory
+    address_arr: usize, // The res of array that push/pop is performed on
+    vartab: &mut Vartable,
+) {
+    // If not empty
+    if let Some(to_add) = cfg.array_lengths_temps.clone().get(&address_arr) {
+        let add_expr = if minus {
+            Expression::Subtract(
+                Loc::Codegen,
+                Type::Uint(32),
+                false,
+                Box::new(Expression::Variable(Loc::Codegen, Type::Uint(32), *to_add)),
+                Box::new(Expression::NumberLiteral(
+                    Loc::Codegen,
+                    Type::Uint(32),
+                    BigInt::one(),
+                )),
+            )
+        } else {
+            Expression::Add(
+                Loc::Codegen,
+                Type::Uint(32),
+                false,
+                Box::new(Expression::Variable(Loc::Codegen, Type::Uint(32), *to_add)),
+                Box::new(Expression::NumberLiteral(
+                    Loc::Codegen,
+                    Type::Uint(32),
+                    BigInt::one(),
+                )),
+            )
+        };
+
+        // Add instruction to the cfg
+        cfg.add(
+            vartab,
+            Instr::Set {
+                loc: Loc::Codegen,
+                res: *to_add,
+                expr: add_expr,
+            },
+        );
+    }
+}
+
+pub(crate) fn handle_array_assign(
+    right: &ast::Expression,
+    cfg: &mut ControlFlowGraph,
+    vartab: &mut Vartable,
+    pos: &usize,
+) {
+    if let ast::Expression::Variable(_, _, right_res) = *right {
+        // If we have initialized a temp var for this var
+        if let Some(to_update) = cfg.array_lengths_temps.clone().get(&right_res) {
+            let num = Expression::Variable(Loc::Codegen, Type::Uint(32), *to_update);
+            let temp_res = vartab.temp_name("array_size", &Type::Uint(32));
+            cfg.add(
+                vartab,
+                Instr::Set {
+                    loc: Loc::Codegen,
+                    res: temp_res,
+                    expr: num,
+                },
+            );
+
+            cfg.array_lengths_temps.insert(*pos, temp_res);
+        } else {
+            // If the right hand side doesn't have a temp, it must be a function parameter.
+            cfg.array_lengths_temps.remove(pos);
+        }
+    } else if let ast::Expression::AllocDynamicArray(_, ty, size, _) = right {
+        let a = if cfg.array_lengths_temps.get(pos) != None {
+            // If there was a temp variable, reassign it to the new value
+            cfg.array_lengths_temps.remove(pos).unwrap()
+        } else {
+            //create a new temp variable
+            vartab.temp_name("array_size", ty)
+        };
+
+        if let ast::Expression::Variable(_, _, position) = **size {
+            let num = Expression::Variable(Loc::Codegen, Type::Uint(32), position);
```

Comment:
`size` is already a `Expression::Variable(Loc::Codegen, Type::Uint(32), position);`. Why are you constructing it again?

Reply:
Size is of type ast::Expression, I'm constructing a codegen::Expression to insert in the cfg